autotest/utilities/test_ogr2ogr_lib.py

```diff
@@ -603,3 +602,24 @@ def test_ogr2ogr_lib_sql_filename():
     lyr = ds.GetLayer(0)
     assert lyr.GetFeatureCount() == 10
     assert lyr.GetLayerDefn().GetFieldIndex('literalfield') == 0
+
+
+###############################################################################
+# Verify -emptyStrAsNull
+
+def test_ogr2ogr_emptyStrAsNull():
+
+    src_ds = gdal.GetDriverByName('Memory').Create('', 0, 0, 0)
+    lyr = src_ds.CreateLayer('layer')
+    lyr.CreateField(ogr.FieldDefn('foo'))
+    f = ogr.Feature(lyr.GetLayerDefn())
+    f['foo'] = ''
+    lyr.CreateFeature(f)
+
+    ds = gdal.VectorTranslate('', src_ds, format='Memory', emptyStrAsNull=True)
+
+    lyr = ds.GetLayer(0)
+    f = lyr.GetNextFeature()
+
+    assert f['foo'] == None, 'expected empty string to be transformed to null'
```

Comment:
Probably slightly more idiomatic:
```suggestion
    assert f['foo'] is None, 'expected empty string to be transformed to null'
```